src/pybind/mgr/coleoid/module.py

```diff
@@ -0,0 +1,432 @@
+"""
+A RESTful API for Ceph
+"""
+
+import json
+import inspect
+import StringIO
+import threading
+import traceback
+import ConfigParser
+
+import api
+import common
+
+from uuid import uuid4
+from flask import Flask
+from OpenSSL import SSL
+from flask_restful import Api
+
+from mgr_module import MgrModule, CommandResult
+
+# Global instance to share
+instance = None
+
+
+
+class CommandsRequest(object):
+    """
+    This class handles parallel as well as sequential execution of
+    commands. The class accept a list of iterables that should be
+    executed sequentially. Each iterable can contain several commands
+    that can be executed in parallel.
+
+    Example:
+    [[c1,c2],[c3,c4]]
+     - run c1 and c2 in parallel
+     - wait for them to finish
+     - run c3 and c4 in parallel
+     - wait for them to finish
+    """
+
+
+    @staticmethod
+    def run(commands, uuid=str(uuid4())):
```

Comment:
This currently gets self.uuid passed in always, so no default needed, and instead of a static method that takes a uuid argument, this could be a class method that just uses self.uuid.